src/util/Algorithm.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <concepts>
+#include <vector>
+
+namespace ad_utility {
+
+template <class T>
+inline bool contains(std::vector<T> elements, T element) {
```

Comment:
1. No need for `inline` here, templates are `inline` implicitly.
2. Both arguments should be `const &`, No need for either of the copies.
3. The limitation to `vector<T>` is a little arbitrary. it could be simply
```
template <typename Container, typename T>
bool contains(const Container& container, const T& element);
```
If you are really motivated you can also use concepts here that the first one is indeed some kind of "container/view/range" and the second one matches the value type.